lib/matplotlib/widgets.py

```diff
@@ -600,12 +607,29 @@ def set_active(self, index):
         for cid, func in self.observers.items():
             func(self.labels[index].get_text())
 
+    def clear(self):
+        """Clears all the checkboxes"""
+
+        for l1, l2 in self.lines:
+            l1.set_visible(False)
+            l2.set_visible(False)
+
+        if self.drawon:
+            self.ax.figure.canvas.draw()
+
     def get_status(self):
         """
         returns a tuple of the status (True/False) of all of the check buttons
         """
         return [l1.get_visible() for (l1, l2) in self.lines]
 
+    def get_checked(self):
+        """Returns a list of labels currently checked by user."""
```

Comment:
Not sure how universally useful this method is. Generally, the labels are a user-facing string aimed at readability and not to identify a state. Only the latter use-case would need this method. This is not something usually found in GUI frameworks, because you tie the state to the displayed string; i.e. when changing the displayed string, you would also have to change your code logic.

I tend not to include this in our public API. However, if other reviewers are ok with it, I'd follow along.

Reply:
I didn't mean this to be a state indicator - which is clearly and explicitly covered by `get_status()` with `status` in its name, but as a result once the user is done checking all the applicable boxes. I am using it in github.com/raamana/visualqc and there is a clear need for this.

I am not dogmatic about the name, we could improve the name if needed.

Reply:
Still, you are using the labels as identifiers for the checked elements. This works well for simple cases, and I agree that it's simpler to use than `get_status()`. However, in general labels don't qualify as identifiers. E.g. among other things labels don't have to be unique.

A more universal solution would associate each check box with a unique identifier and only use the label for the visual text. We don't have this at the moment, but current changes should not make it more difficult to implement.

I would be ok to rename this to `get_checked_labels()`.

Reply:
Interesting.

Yes, current implementation allows duplicate labels, as python list allows it. But index into the list sufficiently unique for this purpose. Using ‘set’ would make the implementation more complicated. 

What do you mean by “you are using the labels as identifiers for the checked elements”?

Reply:
AFAICS the use case it that you want to know which of the boxes is checked. You want something that describes (identifies) the checked boxes. You are probably thinking of something like
~~~
btns = CheckButtons(ax, labels=['a', 'b', 'c'])
...
if 'a' in btns.get_checked():
~~~
You use 'a' as an identifier for the first check box, which has label 'a'. This works in the simple example.

However, in general you cannot infer/identify the checked boxes from the associated labels because labels don't have to be unique.

Reply:
I see what you mean now. At a high level, "labels don't have to be unique" is a design decision that has been made already. Using python lists as the underlying data structure is a fine decision too. Discussion on whether to impose uniqueness (representing them with a `set` perhaps) would be interesting, and is especially well-suited for RadioButtons. Such a decision might require additional logic to maintain valid instantiation of these widgets.